### PR TITLE
Add TestingDefaultPlatform to avoid duplicate logger shutdown

### PR DESCRIPTION
**Pull request by matthiaso**
Some platform listeners must not be used within another platform (which can happen during tests)

331773

### CODE REVIEW

**Comment by matthiaso** [comment]
**Caution: Do not merge yet, wait until merging is allowed again**

File: org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java
Comment (paolobazzi) [inline org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java:28]: Maybe extend PlatformImplementor directly, as DefaultPlatform is an empty class.

File: org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/runner/RunWithNewPlatform.java
Comment (paolobazzi) [inline org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/runner/RunWithNewPlatform.java:23]: Add a note, indicating that a custom implemented platform used together with this annotation should use TestingPlatform as base class.

File: org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java
Comment (paolobazzi) [inline org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java]: maybe extract class filter/accept method to own method, to facilitate subclassing for own testing platforms

**Review by paolobazzi — APPROVED** [review]
Looks good, see minor improvements.

File: org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/runner/RunWithNewPlatform.java
Comment (matthiaso) [reply org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/runner/RunWithNewPlatform.java:23]: Agreed, added an additional paragraph to the annotation.

File: org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java
Comment (matthiaso) [reply org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java]: Extracted method to new method `acceptBean`

File: org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java
Comment (matthiaso) [reply org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java:28]: Is this a must-have? Wouldn't it be better if we keep the extension to `DefaultPlatform` if maybe at some point in time methods are overwritten/added there? Did not change this code yet.

Comment (paolobazzi) [reply org.eclipse.scout.rt.platform.test/src/main/java/org/eclipse/scout/rt/testing/platform/TestingDefaultPlatform.java:28]: Not a must-have, just a suggestion. Your argument is valid, so keep it extending DefaultPlatform.

**Review by paolobazzi — APPROVED** [review]

**Pull request merged**
